pkg/vm/emit/emit.go

```diff
@@ -160,6 +164,54 @@ func Array(w *io.BinWriter, es ...any) {
 	Opcodes(w, opcode.PACK)
 }
 
+// Convertible converts provided stackitem.Convertible to the stackitem.Item and
+// emits the item to the given buffer.
+func Convertible(w *io.BinWriter, c stackitem.Convertible) {
+	si, err := c.ToStackItem()
+	if err != nil {
+		w.Err = fmt.Errorf("failed to convert stackitem.Convertible to stackitem: %w", err)
+		return
+	}
+	StackItem(w, si)
+}
+
+// StackItem emits provided stackitem.Item to the given buffer.
+func StackItem(w *io.BinWriter, si stackitem.Item) {
+	switch t := si.Type(); t {
+	case stackitem.AnyT:
+		if si.Value() == nil {
+			Opcodes(w, opcode.PUSHNULL)
+		} else {
+			w.Err = fmt.Errorf("only nil value supported for %s", t)
+			return
+		}
+	case stackitem.BooleanT:
+		Bool(w, si.Value().(bool))
+	case stackitem.IntegerT:
+		BigInt(w, si.Value().(*big.Int))
+	case stackitem.ByteArrayT, stackitem.BufferT:
+		Bytes(w, si.Value().([]byte))
+	case stackitem.ArrayT, stackitem.StructT:
+		arr := si.Value().([]stackitem.Item)
+		arrAny := make([]any, len(arr))
+		for i := range arr {
+			arrAny[i] = arr[i]
+		}
+		Array(w, arrAny...)
+	case stackitem.MapT:
+		// TODO: this one is very controversial, I vote for returning an error here.
+		arr := si.Value().([]stackitem.MapElement)
+		arrAny := make([]any, len(arr))
+		for i := range arr {
+			arrAny[i] = &arr[i] // *stackitem.MapElement is Convertible.
+		}
+		Array(w, arrAny...)
```

Comment:
Array is not a map.

Reply:
I've used `PACKSTRUCT` and `PACKMAP` so rarely that I completely forgot that they exist. These two opcodes solve the problem, fixed.